extensions/2.0/Khronos/KHR_materials_anisotropy/README.md

```diff
@@ -0,0 +1,155 @@
+
+| Direction    | Strength 0.5                | Strength 0.0                |
+| ------------ | --------------------------- | --------------------------- |
+| `0.0`        | ![Fig. 3](figures/fig3.jpg) | ![Fig. 2](figures/fig2.jpg) |
+| `0.78539816` | ![Fig. 6](figures/fig6.jpg) | ![Fig. 5](figures/fig5.jpg) |
+| `1.57079633` | ![Fig. 1](figures/fig1.jpg) | ![Fig. 5](figures/fig5.jpg) |
+
+To achieve certain surface finishes, it is possible to define the anisotropy and its direction using a texture.
+
+![Fig. 8](figures/fig8.jpg)
+![Fig. 10](figures/fig10.jpg)
+
+## Implementation
+
+In the following example, `u_AnisotropyStrength` is set to this extension's `anisotropyStrength`, and `u_AnisotropyDirection` is initialized as `[ cos(anisotropyDirection), sin(anisotropyDirection) ]`.
+
+The default value of `anisotropyDirection` is zero, so when this parameter is not supplied by glTF, `u_AnisotropyDirection` will be `[ 1.0, 0.0 ]`.
+
+```glsl
+uniform float u_AnisotropyStrength;
+uniform vec2 u_AnisotropyDirection;
+
+float anisotropy = u_AnisotropyStrength;
+vec2 direction = u_AnisotropyDirection;
+
+#if HAS_ANISOTROPY_MAP
+vec2 anisotropyTexture = texture(uv, u_AnisotropyTextureSampler).rg * 2.0 - vec2(1.0);
+anisotropy *= length(anisotropyTexture);
+direction = mat2(direction.x, -direction.y, direction.y, direction.x) * normalize(anisotropyTexture);
+#endif
+
+vec3 anisotropicT = normalize(TBN * vec3(direction, 0.0));
+vec3 anisotropicB = normalize(cross(normal_geometric, anisotropicT));
+
+float TdotL = dot(anisotropicT, l);
+float BdotL = dot(anisotropicB, l);
+float TdotH = dot(anisotropicT, h);
+float BdotH = dot(anisotropicB, h);
+
+f_specular += intensity * NdotL * BRDF_specularAnisotropicGGX(f0, f90, alphaRoughness,
+    VdotH, NdotL, NdotV, NdotH,
+    BdotV, TdotV, TdotL, BdotL, TdotH, BdotH, anisotropy);
+```
+
+The anisotropic GGX is defined as follows:
+
+```glsl
+vec3 BRDF_specularAnisotropicGGX(vec3 f0, vec3 f90, float alphaRoughness,
+    float VdotH, float NdotL, float NdotV, float NdotH, float BdotV, float TdotV,
+    float TdotL, float BdotL, float TdotH, float BdotH, float anisotropy)
+{
+    float at = max(alphaRoughness * (1.0 + anisotropy), 0.00001);
+    float ab = max(alphaRoughness * (1.0 - anisotropy), 0.00001);
+
+    vec3 F = F_Schlick(f0, f90, VdotH);
+    float V = V_GGX_anisotropic(NdotL, NdotV, BdotV, TdotV, TdotL, BdotL, anisotropy, at, ab);
+    float D = D_GGX_anisotropic(NdotH, TdotH, BdotH, anisotropy, at, ab);
+
+    return F * V * D;
+}
+
+float D_GGX_anisotropic(float NdotH, float TdotH, float BdotH, float anisotropy, float at, float ab)
+{
+    float a2 = at * ab;
+    vec3 f = vec3(ab * TdotH, at * BdotH, a2 * NdotH);
+    float w2 = a2 / dot(f, f);
+    return a2 * w2 * w2 / M_PI;
+}
+
+float V_GGX_anisotropic(float NdotL, float NdotV, float BdotV, float TdotV, float TdotL, float BdotL,
+    float anisotropy, float at, float ab)
```

Comment:
`anisotropy` is an unused parameter in both these functions - `at` and `ab` already represent its effect.